cpp/log/cert.cc

```diff
@@ -606,27 +613,27 @@ Cert::Status Cert::ExtensionIndex(int extension_nid,
                << " for NID " << extension_nid
                << ". Is the NID not recognised?";
     LOG_OPENSSL_ERRORS(ERROR);
-    return ERROR;
+    return util::Status(util::error::INTERNAL, "X509_get_ext_by_NID error");
   }
   if (index == -1)
-    return FALSE;
-  *extension_index = index;
-  return TRUE;
+    return util::Status(util::error::NOT_FOUND, "extension not found");
+  return index;
 }
 
 
 Cert::Status Cert::GetExtension(int extension_nid,
                                 X509_EXTENSION** ext) const {
-  int extension_index;
-  Status status = ExtensionIndex(extension_nid, &extension_index);
-  if (status != TRUE) {
-    return status;
+  const StatusOr<int> extension_index(ExtensionIndex(extension_nid));
+  if (!extension_index.ok()) {
+    return extension_index.status().CanonicalCode() == util::error::NOT_FOUND
```

Comment:
`CanonicalCode()` is preferable, as it returns the `enum` (`error_code()` simply returns `int`). This can make a difference in a `switch` statement (where compiler can catch unhandled `enum` values), but otherwise, not that much.